src/Microsoft.OpenApi.OData.Reader/OpenApiExtensions/OpenApiEnumValuesDescriptionExtension.cs

```diff
@@ -0,0 +1,71 @@
+// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// ------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.OpenApi.Interfaces;
+using Microsoft.OpenApi.OData.Common;
+using Microsoft.OpenApi.Writers;
+
+namespace Microsoft.OpenApi.OData.OpenApiExtensions;
+
+/// <summary>
+/// Extension element for OpenAPI to add enum values descriptions.
+/// Based of the AutoRest specification https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-enum
+/// </summary>
+internal class OpenApiEnumValuesDescriptionExtension : IOpenApiExtension
+{
+	/// <summary>
+    /// Name of the extension as used in the description.
+    /// </summary>
+	public string Name => "x-ms-enum";
+	/// <summary>
+	/// The of the enum.
+	/// </summary>
+	public string EnumName { get; set; }
+	/// <summary>
+	/// Descriptions for the enum symbols, where the value MUST match the enum symbols in the main description
+	/// </summary>
+	public List<EnumDescription> ValuesDescriptions { get; set; } = new();
```

Comment:
Doesn't need the "set" for the collection since you have new(), right?

Reply:
the set is here so people can assign the whole collection (used in unit tests)